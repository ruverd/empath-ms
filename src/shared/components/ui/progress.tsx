import * as ProgressPrimitive from "@radix-ui/react-progress";
import * as React from "react";

import { cn } from "@/shared/lib/utils";

const Progress = React.forwardRef<
  React.ElementRef<typeof ProgressPrimitive.Root>,
  React.ComponentPropsWithoutRef<typeof ProgressPrimitive.Root>
>(({ className, value, ...props }, ref) => (
  <ProgressPrimitive.Root
    ref={ref}
    className={cn(
      "relative h-4 w-full overflow-hidden rounded-full bg-shades-violet-2",
      className
    )}
    {...props}
  >
    <ProgressPrimitive.Indicator
      className="h-full w-full flex-1 rounded-full transition-all"
      style={{
        background: "linear-gradient(90deg, #5C15A7 0%, #0086A6 94.17%)",
        width: `${value}%`,
      }}
    />
  </ProgressPrimitive.Root>
));
Progress.displayName = ProgressPrimitive.Root.displayName;

export { Progress };
