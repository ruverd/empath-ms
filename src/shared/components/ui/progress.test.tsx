import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import { Progress } from "./progress";

describe("Progress", () => {
  it("should match snapshot", () => {
    const { container } = render(<Progress value={50} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should renders with default value", () => {
    render(<Progress value={50} />);
    const progressBar = screen.getByRole("progressbar");
    expect(progressBar).toBeInTheDocument();
  });

  it("should applies custom className", () => {
    render(<Progress value={50} className="custom-class" />);
    const progressBar = screen.getByRole("progressbar");
    expect(progressBar).toHaveClass("custom-class");
  });
});
