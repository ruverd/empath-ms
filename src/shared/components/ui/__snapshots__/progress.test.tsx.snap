// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Progress > should match snapshot 1`] = `
<div
  aria-valuemax="100"
  aria-valuemin="0"
  class="relative h-4 w-full overflow-hidden rounded-full bg-shades-violet-2"
  data-max="100"
  data-state="indeterminate"
  role="progressbar"
>
  <div
    class="h-full w-full flex-1 rounded-full transition-all"
    data-max="100"
    data-state="indeterminate"
    style="width: 50%;"
  />
</div>
`;
